protostar/cli/argument_value_from_config_extractor.py

```diff
@@ -0,0 +1,65 @@
+from typing import Any, Optional, Protocol
+
+
+class ArgumentValueFromConfigProviderProtocol(Protocol):
```

Comment:
Do we need this protocol? How many classes implement it?

Reply:
This protocol is extended by the `ConfigurationFile` protocol. The purpose of this Protocol is inverting dependency.

from: `protostar.cli` -> `protostar.configuration_file`
to: `protostar.cli` <- `protostar.configuration_file`

Reply:
Can you please address my question? 

> How many classes implement it?

Reply:
This protocol is extended by the `ConfigurationFile` protocol. And the `ConfigurationFile` protocol is implemented by two classes.

Reply:
So why the abstraction extist if only one class extends it?

Reply:
[use ArgumentValueResolver](https://github.com/software-mansion/protostar/pull/901/commits/84905fb42a7b05d3106fda89e5c0013dbdd99c5a)

I removed that protocol. This approach can potentially bite in the future.

Reply:
I don't think we have to abstract protocol over that

Reply:
Resolving in favor of the new thread